delfin/task_manager/tasks/resources.py

```diff
@@ -397,6 +397,159 @@ def remove(self):
         db.disk_delete_by_storage(self.context, self.storage_id)
 
 
+class StorageFilesystemTask(StorageResourceTask):
+    def __init__(self, context, storage_id):
+        super(StorageFilesystemTask, self).__init__(context, storage_id)
+
+    @check_deleted()
+    @set_synced_after()
+    def sync(self):
```

Comment:
Sync will be called for each subclass of StorageResourceTask ! how do we make sure it is called only if it the driver is an instance of NAS driver ?